src/app/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react';
+
+export default function useLocalStorage<T>(
+  key: string,
+  initialValue: T
+): [storedValue: T, setValue: (value: T) => void] {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    const item = localStorage.getItem(key);
+    return item ? JSON.parse(item) : initialValue;
+  });
```

Comment:
Here's an example of the same hook but with `try... catch` 
https://github.com/marwinburesch/moviescout/blob/main/src/app/hooks/useLocalStorage.ts